[Bug]: Worker placement dispatch coalesces different execution modes

## Bug type

Behavior bug (incorrect output/state without crash)

## Beta release blocker

No

## Summary

Concurrent cloud-worker placement dispatch requests for the same session are incorrectly deduplicated when one requests `worker-turn` and the other requests `remote-exec`.

## Steps to reproduce

1. Start a placement dispatch for a session in `worker-turn` mode and hold the operation in flight.
2. Dispatch the same session/profile request again with `executionMode: "remote-exec"`.
3. Observe that the second request joins and resolves with the first operation instead of being rejected as a different request.

Focused current-main reproduction: `node scripts/run-vitest.mjs src/gateway/worker-environments/placement-dispatch-coordinator.test.ts` failed before the repair because the conflicting promise resolved with `{ state: "active" }` instead of rejecting.

## Expected behavior

The coordinator may deduplicate only semantically identical placement dispatch requests. Requests with different execution modes must conflict because `worker-turn` and `remote-exec` use different placement and turn-claim authority.

## Actual behavior

The in-flight identity comparison checks session key, agent, profile, device, and inherited provider profile, but omits `executionMode`, so the second mode can share the first operation.

## OpenClaw version

Current `main` at `9d6b64cabf28618f74b6868e9ca2cb8ffa5c8dec`.

## Operating system

macOS 26 (source checkout); reproduced and validated on Linux through Blacksmith Testbox.

## Install method

Source checkout.

## Model

Not model-dependent.

## Provider / routing chain

Gateway placement dispatch coordinator; not provider-dependent.

## Logs

```shell
AssertionError: promise resolved "{ state: 'active' }" instead of rejecting
```

## Screenshots, recordings, and evidence

The focused regression fails on pre-fix code and passes after adding `executionMode` to the coordinator-owned in-flight request identity. The isolated Gateway/worker lifecycle test also passes through restart, workspace sync, and safe session retirement.

## Impact and severity

- Affected: concurrent Gateway cloud-worker placement dispatches for one session when its effective runtime mode changes.
- Severity: bounded authority/lifecycle correctness bug; no protocol, persistence, config, or public API change is required.
- Frequency: deterministic for overlapping requests that differ only by execution mode.
- Consequence: a caller can receive another authority mode's in-flight placement result instead of a conflict and retry outcome.

## Additional information

The missing field was introduced when `executionMode` became part of `WorkerPlacementDispatchRequest`; the older exact-request comparator was not updated. The canonical fix is one comparison in the coordinator plus a regression in the existing coalescing test. Production LOC +1; tests +4.

--- a/src/gateway/worker-environments/placement-dispatch-coordinator.test.ts
+++ b/src/gateway/worker-environments/placement-dispatch-coordinator.test.ts
@@ -63,9 +63,13 @@ describe("worker placement dispatch coordinator", () => {
         },
       }),
     ).rejects.toThrow(`Session ${REQUEST.sessionKey} is already dispatching another request`);
+    const modeConflict = expect(
+      coordinated.dispatch({ ...REQUEST, executionMode: "remote-exec" }),
+    ).rejects.toThrow(`Session ${REQUEST.sessionKey} is already dispatching another request`);
     const retry = coordinated.dispatch(REQUEST);
     releaseDispatch.resolve();
 
+    await modeConflict;
     const [firstResult, retryResult] = await Promise.all([first, retry]);
     expect(retryResult).toBe(firstResult);
     expect(dispatch).toHaveBeenCalledOnce();

--- a/src/gateway/worker-environments/placement-dispatch-coordinator.ts
+++ b/src/gateway/worker-environments/placement-dispatch-coordinator.ts
@@ -91,6 +91,7 @@ export function coordinateWorkerPlacementDispatch(
           inFlight.request.sessionKey !== request.sessionKey ||
           inFlight.request.agentId !== request.agentId ||
           inFlight.request.profileId !== request.profileId ||
+          inFlight.request.executionMode !== request.executionMode ||
           inFlight.request.deviceId !== request.deviceId ||
           !isDeepStrictEqual(inFlight.request.inheritedProfile, request.inheritedProfile)
         ) {

